src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -2469,31 +2469,37 @@ ITypeSymbol ITypeSymbolInternal.GetITypeSymbol()
 
         internal abstract bool HasInlineArrayAttribute(out int length);
 
-        internal TypeWithAnnotations TryGetInlineArrayElementType()
+#nullable enable
+        internal FieldSymbol? TryGetInlineArrayElementField()
         {
             Debug.Assert(HasInlineArrayAttribute(out var length) && length > 0);
 
-            TypeWithAnnotations elementType = default;
+            FieldSymbol? elementField = null;
 
             if (this.TypeKind == TypeKind.Struct)
             {
-                foreach (Symbol member in GetMembers())
+                foreach (FieldSymbol field in ((NamedTypeSymbol)this).OriginalDefinition.GetFieldsToEmit())
                 {
-                    if (member is FieldSymbol { IsStatic: false } field)
+                    if (!field.IsStatic)
                     {
-                        if (field.RefKind != RefKind.None || elementType.HasType)
+                        if (field.RefKind != RefKind.None || elementField is not null)
                         {
-                            return default;
+                            return null;
                         }
                         else
                         {
-                            elementType = field.TypeWithAnnotations;
+                            elementField = field;
                         }
                     }
                 }
             }
 
-            return elementType;
+            if (elementField is not null && elementField.ContainingType.IsGenericType)
```

Comment:
>elementField.ContainingType.IsGenericType [](http://example.com/codeflow?start=44&length=41)

Are we testing nested types, with and without generic type parameters?

Reply:
> Are we testing nested types, with and without generic type parameters?

We are not at the moment, but ```IsGenericType``` checks covers the scenario. I will add a test.

Reply:
Added tests.